GET dashboard question candidates

johnswanson opened the pull request:
Works for both the root and normal collections.

johnswanson committed:
GET dashboard question candidates

Works for both the root and normal collections.

camsaul commented on test/metabase/api/collection_test.clj:
```diff
@@ -1494,6 +1494,250 @@
+                    keys
+                    (into #{}))))))
+
+    (testing "Card is in two dashboards"
```
These could easily be several separate tests. If they're not sharing state, it's better to break out logically separate test cases into separate `deftest`s, that way if they fail they're less of a headache to debug and so we can parallelize them further

johnswanson replied:
Good call - thanks!

camsaul commented on test/metabase/api/collection_test.clj:
```diff
@@ -1494,6 +1494,250 @@
+                    first
+                    :sole_dashboard_info
+                    keys
+                    (into #{}))))))
+
+    (testing "Card is in two dashboards"
```
```suggestion
                    (into #{}))))))))

(deftest ^:parallel dashboard-question-candidates-card-is-in-two-dashboards-test
  (testing "GET /api/collection/:id/dashboard-question-candidates"
    (testing "Card is in two dashboards"
```

noahmoss reviewed (approved):
very nice

johnswanson committed:
Attempt to resolve deadlock

johnswanson committed:
Mark tests as non-parallel

@noahmoss and I tried to debug this and were pretty stumped.

On MariaDB 10.2, we're getting extremely flaky results from running
parallel tests that include `mt/with-temp` blocks inside.

johnswanson committed:
Split up move dashboard question tests
